adding value for "taking next assessment" to the "completed" event!

amitsinghsutara opened the pull request:
# Changes
- adding value for "taking next assessment" to the "completed" event!



Ref: [AJ-394](https://curiouslearning.atlassian.net/browse/aj-394)


[AJ-394]: https://curiouslearning.atlassian.net/browse/AJ-394?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

amitsinghsutara committed:
Renaming Folder!

amitsinghsutara committed:
Revert "Renaming Folder!"

This reverts commit c002d54e151cb1f021fce55dcd9051a4876f5478.

amitsinghsutara committed:
adding value for "taking next assessment" to the "completed" event

dz4va commented on src/utils/urlUtils.ts:
```diff
@@ -38,7 +38,12 @@ export function getDataFile(): string {
 
+  let pathParams = getPathName();
+  let requiredScore = Number(pathParams.get("requiredScore"));
+  return requiredScore >= 640
```
This is wrong right here, @amitsinghsutara 
We can't just compare the required score to 640 which is a hardcoded value.

Once we get the score and we are about to send the completed event we have to compare the requiredScore from the path params to the assessment score for Letter Sounds that the user gets at the end of Letter Sounds and send a bool if the nextAssessment is null or not 🙌🏽

dz4va commented on src/analytics/analyticsEvents.ts:
```diff
@@ -295,24 +296,46 @@ export class AnalyticsEvents {
+        appVersion: AnalyticsEvents.appVersion,
+        contentVersion: AnalyticsEvents.contentVersion,
+      });
+    }
```
Instead of having 2 similar blocks we can just send the nextAssessment NULL or other value, which we could possibly provide from the Synapse? Like the next assessment? hausa-sight-words as an example?

also too much code duplication here

amitsinghsutara committed:
Updating logic for sending nextAssessment values!

:- Use a single eventData object to make it easier to read and maintain.

:-Use the spread operator ... to conditionally add properties (synapseAccess and nextAssessment) only if isSynapseUser is true.

amitsinghsutara committed:
additional fixes!

amitsinghsutara committed:
Additional OPtimizations!

dz4va commented on src/utils/urlUtils.ts:
```diff
@@ -38,6 +38,20 @@ export function getDataFile(): string {
+
+  let pathParams = getPathName();
+  let requiredScore = Number(pathParams.get("requiredScore"));
+  return requiredScore >= 640
```
Let's remove the isItLastAssessment routine, as it's not being used anymore? @amitsinghsutara

amitsinghsutara committed:
Removing unused imports!
